### PR TITLE
Use function definitions instead of lambdas

### LINKED ISSUE #102: Fix pep8 violations according to new 1.6.0 version
There was a new release of pep8 yesterday (Feb 6th, 2015) that added some new checks that pep257 violates (most notably lambda expression assignment). We need to fix these violations for our build to pass.

### PR DESCRIPTION
**Pull request by sigmavirus24**
Fixes #102

### CODE REVIEW

**Comment by sigmavirus24** [comment]
So it looks as if whatever plugin you're using to run pep8 is doing the equivalent of

```
$ pep8 --ignore=E701
```

But `E704` is off by default. Should we add `E704` to the ignore list as well?

File: pep257.py [hunk 1]
```diff
@@ -54,22 +54,43 @@ def next(obj, default=nothing):
...
+    def __eq__(self, other):
+        return other and vars(self) == vars(other)
 
     def __repr__(self):
-        format_arg = lambda arg: '{}={!r}'.format(arg, getattr(self, arg))
-        kwargs = ', '.join(format_arg(arg) for arg in self._fields)
+        kwargs = ', '.join(
+            map(lambda arg: '{}={!r}'.format(arg, getattr(self, arg)),
+                self._fields)
+            )
```
Comment (Nurdok) [inline pep257.py]: I think this change just makes the code both longer and less readable.

**Comment by Nurdok** [comment]
I left one comment. Otherwise, this is great. Thank you so much for your time and effort!

File: pep257.py (hunk 1 above)
Comment (jirikuncar) [inline pep257.py]: What about something like:
```python
kwargs = ','.join('{}={!r}'.format(arg, getattr(self, arg)) for arg in self._fields)
```

File: pep257.py [hunk 2]
```diff
@@ -100,7 +123,9 @@ class Module(Definition):
     _nest = staticmethod(lambda s: {'def': Function, 'class': Class}[s])
     module = property(lambda self: self)
     all = property(lambda self: self._all)
```
Comment (keleshev) [inline pep257.py]: Ha-ha, lambda assignments are not allowed, but property and staticmethod assignments are fine, amirite? :smile:

File: pep257.py [hunk 3]
```diff
@@ -100,7 +123,9 @@ class Module(Definition):
...
-    __str__ = lambda self: 'at module level'
+
+    def __str__(self):
+        return 'at module level'
```
Comment (keleshev) [inline pep257.py]: Thanks, PEP 8, this is now much more readable! :smile_cat:

Comment (sigmavirus24) [inline pep257.py]: It's been in the document for years now, it just wasn't checked by the tool.

File: pep257.py (hunk 2 above)
Comment (sigmavirus24) [inline pep257.py]: I'm happy to change those, but the goal was to fix the test failures, not make everything more conventional.

File: pep257.py (hunk 1 above)
Comment (sigmavirus24) [inline pep257.py]: @jirikuncar I'm pretty sure that'll trigger a line to long error which I think @Nurdok will still object to when formatted properly. Regardless, let me see how that looks after accounting for the default maximum line length.

Comment (sigmavirus24) [inline pep257.py]: Handled in dc77669

**Comment by Nurdok** [comment]
Thanks @sigmavirus24 !

**Pull request merged**